kahuna/public/js/search-query/service.js

```diff
@@ -0,0 +1,32 @@
+import angular from 'angular';
+import Rx from 'rx';
+import '../util/rx';
+
+export const searchQueryService = angular.module('gr.searchQuery.service', ['util.rx']);
+
+searchQueryService.factory('searchQueryService', [function() {
+    const query$ = new Rx.Subject();
+
+    // TODO: When we update rx, we need to swap the seed / acc around
+    const q$ = query$.scan('', (state, fn) => fn(state).trim());
```

Comment:
Also: check the [other reactive state helpers](https://github.com/guardian/grid/blob/master/kahuna/public/js/lib/data-structure/list-factory.js#L17-L19), I suspect you may want to add `distinctUntilChanged` (to skip identical values) and `shareReplay` (to share subscription and replay the last value to new subscribers).